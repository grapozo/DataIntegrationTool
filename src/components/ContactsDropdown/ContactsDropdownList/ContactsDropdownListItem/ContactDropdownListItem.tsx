import styles from './ContactsDropdownListItem.scss';

export const ContactDropdownListItem = (): JSX.Element => {
  return (
    <div className={`${styles.contactDropdownListItem}`}>
      <label>
        <input type="checkbox" />
      </label>
    </div>
  );
};
