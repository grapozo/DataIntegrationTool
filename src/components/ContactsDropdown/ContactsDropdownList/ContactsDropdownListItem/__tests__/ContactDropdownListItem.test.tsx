import { create } from 'react-test-renderer';
import { Contact } from 'shared/interfaces/contact.interface';
import { ContactDropdownListItem } from '../ContactDropdownListItem';
import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

const mockContact: Contact = {
  id: '9ba3f8f7d420c5fb33d7fbf2704e24fe',
  name: 'Mock contact'
};

const mockOnContactSelectCallback = jest.fn();

const MockDropdownListItem = (): JSX.Element => {
  return (
    <ContactDropdownListItem
      contact={mockContact}
      onContactSelectCallback={mockOnContactSelectCallback}
    />
  );
};

describe('ContactDropdownListItem', () => {
  it('should correctly render', () => {
    const tree = create(<MockDropdownListItem />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should check the checkbox when the contact name is clicked', () => {
    render(<MockDropdownListItem />);
    const checkbox = screen.getByLabelText(mockContact.name) as HTMLInputElement;
    fireEvent.click(checkbox);
    expect(checkbox).toBeChecked();
  });
  it('should call onContactSelectCallback when the checkbox is selected', () => {
    render(<MockDropdownListItem />);
    fireEvent.click(screen.getByLabelText(mockContact.name));
    expect(mockOnContactSelectCallback).toHaveBeenCalled();
  });
});
