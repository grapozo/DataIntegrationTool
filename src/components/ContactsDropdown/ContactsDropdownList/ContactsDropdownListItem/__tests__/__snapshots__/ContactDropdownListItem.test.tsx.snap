// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactDropdownListItem should correctly render 1`] = `
<div
  className="undefined"
>
  <label>
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
    />
    Mock contact
  </label>
</div>
`;
